### PR TITLE
Created telset for remote-storage component 

### PR DESCRIPTION
**Pull request by Wise-Wizard**
**Which problem is this PR solving?**

This PR addresses a part of the issue [#5633 ](https://github.com/jaegertracing/jaeger/issues/5633)

**Description of the changes**
This is a Draft PR to achieve Observability Parity between V1 and V2 components by creating an unified telemetery container to pass observability clients to V1 components.
**How was this change tested?**

The changes were tested by running the following command:

```bash
make test
```

**Checklist**

- [x] I have read [CONTRIBUTING_GUIDELINES.md](https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md)
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - `for jaeger: make lint test`
  - `for jaeger-ui: yarn lint` and `yarn test`

### CODE REVIEW

**Review by yurishkuro — COMMENTED** [review]
please name PRs according to the contributing guidelines

**Comment by Wise-Wizard** [comment]
PTAL and lmk if any changes that are requuired.
Thanks!

File: cmd/remote-storage/app/server.go [hunk 1]
```diff
@@ -129,15 +128,15 @@ func (s *Server) Start() error {
...
 	s.grpcConn = listener
 	s.wg.Add(1)
 	go func() {
 		defer s.wg.Done()
 		if err := s.grpcServer.Serve(s.grpcConn); err != nil {
-			s.logger.Error("GRPC server exited", zap.Error(err))
+			s.Logger.Error("GRPC server exited", zap.Error(err))
+			s.ReportStatus(component.NewFatalErrorEvent(err))
 		}
-		s.healthcheck.Set(healthcheck.Unavailable)
```
Comment (yurishkuro) [inline cmd/remote-storage/app/server.go:140]: send Stopped event from Close()?

File: cmd/remote-storage/app/server_test.go [hunk 2]
```diff
@@ -51,14 +52,16 @@ func TestNewServer_CreateStorageErrors(t *testing.T) {
 	factory.On("CreateSpanWriter").Return(nil, nil)
 	factory.On("CreateDependencyReader").Return(nil, errors.New("no deps")).Once()
 	factory.On("CreateDependencyReader").Return(nil, nil)
-
+	telset := telemetery.Setting{
+		Logger:       zap.NewNop(),
+		ReportStatus: telemetery.HCAdapter(healthcheck.New()),
```
Comment (yurishkuro) [inline cmd/remote-storage/app/server_test.go:57]: if you are not using status reporter, don't pass real function
```suggestion
		ReportStatus: func(*component.StatusEvent) {},
```

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**